### DISCUSSION TITLE
inline vs inline static variable in namespace.

### DISCUSSION THREAD

**Discussion by lonelybud** [YimMenu/YimMenu#2181] [Q&A]
what is the use and differences in `inline` and `inline static` variables. Also how to make sure that variable inside namespace which is included in multiple source files is declared once and reused by compiler. Why there is no use of `extern` in menu code?

example -
```C++

// name.hpp
namespace name {
  int myVar = 1;
}

// main1.cpp
namespace name {
  int myVar = 1;
}

// main2.cpp
namespace name {
  name.myVar = 1;
}
```

## Main reason for understanding
ensure this error does not come up `error LNK2005: "int  name::myVar " already defined in main1.cpp.obj`
and also ensure that variable is not duplicated across multiple source files.

Edit -
According to research `inline` solves the problem but i think it will duplicate the variable across different source files

**Comment by pierrelasse** [comment]
![image](https://github.com/YimMenu/YimMenu/assets/76223390/23f4b3d1-edde-4f54-be8b-e4cc277f59a9)
![image](https://github.com/YimMenu/YimMenu/assets/76223390/61ad8dfc-c803-4b33-9e23-f941395a0f4b)
![image](https://github.com/YimMenu/YimMenu/assets/76223390/018ea423-1178-4830-9a5d-8996fed940e4)
![image](https://github.com/YimMenu/YimMenu/assets/76223390/c25bb4a3-7cdf-4aed-b422-d7090f307908)

: - D

**Reply by lonelybud** [reply]
i believe chat gpt is right but I need answer from experienced devs. So `inline static` should make the code duplicate across multiple source files?
